compiler/vmdef.nim

```diff
@@ -261,11 +261,34 @@ type
     config*: ConfigRef
     graph*: ModuleGraph
     oldErrorCount*: int
+    profiler*: Profiler
+
+  PStackFrame* = ref TStackFrame
+  TStackFrame* = object
+    prc*: PSym                 # current prc; proc that is evaluated
+    slots*: seq[TFullReg]      # parameters passed to the proc + locals;
+                              # parameters come first
+    next*: PStackFrame         # for stacking
+    comesFrom*: int
+    safePoints*: seq[int]      # used for exception handling
+                              # XXX 'break' should perform cleanup actions
+                              # What does the C backend do for it?
+  ProfileInfo* = object
+    time*: float
+    count*: int
```

Comment:
would be nice to add a field `procname` so that you can show the proc name in the profile dump instead of just lineinfo

Reply:
Where would I find that info?

Reply:
there is `TStackFrame.prc.name.s` and `TCtx.prc.sym.name.s`

Reply:
Ok, will look into that.

Reply:
This will use more memory, or add more complexity to the profiler, which I'm not sure is worth the convenience.